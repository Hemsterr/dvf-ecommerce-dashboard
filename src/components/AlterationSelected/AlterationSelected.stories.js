// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import AlterationItem from './index'

storiesOf('AlterationItem', module).add('Default', () => (
  <AlterationItem alteration="Ciara Satin Crepe Pants" />
))
