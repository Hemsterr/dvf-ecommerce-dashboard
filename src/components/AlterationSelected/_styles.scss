.alteration {
  border: pxToRem(2px) solid $big-stone;
  width: pxToRem(300px);
  font-family: $font-base-futura;

  &__wrapper {
    background: $big-stone;
    color: $alabaster;
    @include text-size(pxToRem(12px), pxToRem(3.6px), pxToRem(16px));
    height: pxToRem(50px);
    display: flex;
    justify-content: center;
    align-items: center;

    &:last-child {
      justify-content: normal;
      padding-left: pxToRem(37px);
    }
  }

  &__item {
    padding: pxToRem(40px) pxToRem(15px) 0 pxToRem(20px);
    display: flex;
    justify-content: space-between;

    &:last-child {
      padding-bottom: pxToRem(30px);
    }
  }

  &__selected {
    height: pxToRem(260px);
    overflow: auto;
  }

  &__price {
    @include text-size(pxToRem(12px), pxToRem(2px), pxToRem(20px));
    color: $black;
  }
}


@media only screen and (max-width: $screen-xs) {
  .alteration  {
    &__wrapper {
      @include text-size(pxToRem(10px), pxToRem(2px), pxToRem(13px));
      height: pxToRem(30px);
    }

    &__item {
      padding: pxToRem(20px) pxToRem(15px) 0 pxToRem(20px);
    }

    &__selected {
      height: pxToRem(170px);
    }
  }
}
