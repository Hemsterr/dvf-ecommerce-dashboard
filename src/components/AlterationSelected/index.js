// @flow
import React from 'react'

// Components
import CheckBox from '../CheckBox'

type Props = {
  alterations: Array<{
    id: string,
    label: string,
    isChecked: boolean,
    price: number,
  }>,
}

class AlterationSelected extends React.PureComponent<Props> {
  static defaultProps = {
    alterations: [],
  }

  render() {
    const { alterations } = this.props
    let price = 0
    let count = 0
    for (let i = 0; i < alterations.length; i++) {
      if (alterations[i].isChecked) {
        price += alterations[i].price
        count += 1
      }
    }

    return (
      <div className="alteration">
        <div className="alteration__wrapper">
          <p>Choose Alteration(s)</p>
        </div>
        <div className="alteration__selected">
          {alterations.map(item => (
            <div className="alteration__item" key={item.id}>
              <CheckBox label={item.label} isChecked={item.isChecked} />
              <p className="alteration__price">{`${item.price}$`}</p>
            </div>
          ))}
        </div>
        <div className="alteration__wrapper">
          <p>{`Estimated Quote: ${count ? '$' + price : ''}`}</p>
        </div>
      </div>
    )
  }
}

export default AlterationSelected
