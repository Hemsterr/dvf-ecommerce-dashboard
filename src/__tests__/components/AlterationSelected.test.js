// Libs
import React from 'react'
import { render } from '@testing-library/react'

// Component
import AlterationSelected from '../../components/AlterationSelected'

const mockProps = {
  alterations: [
    {
      label: 'Shorten Sleeves',
      isChecked: true,
      price: 10,
    },
    {
      label: 'Taper Waist',
      isChecked: false,
      price: 60,
    },
  ],
}

const wrapper = (props = {}) => {
  const component = render(<AlterationSelected {...props} />)

  return component
}

describe('AlterationSelected component', () => {
  // Snapshot
  test('should render correctly AlterationSelected component default props', () => {
    const { container } = wrapper()
    expect(container).toMatchSnapshot()
    expect(AlterationSelected.defaultProps.alterations.length).toEqual(0)
  })

  // Testing props
  test('should be rendered with props', () => {
    const { container } = wrapper(mockProps)
    expect(container.querySelectorAll('.alteration__item').length).toEqual(
      mockProps.alterations.length
    )
  })
})
