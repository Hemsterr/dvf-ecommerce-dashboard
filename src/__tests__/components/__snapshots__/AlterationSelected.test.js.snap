// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlterationSelected component should render correctly AlterationSelected component default props 1`] = `
<div>
  <div
    class="alteration"
  >
    <div
      class="alteration__wrapper"
    >
      <p>
        Choose Alteration(s)
      </p>
    </div>
    <div
      class="alteration__selected"
    />
    <div
      class="alteration__wrapper"
    >
      <p>
        Estimated Quote: 
      </p>
    </div>
  </div>
</div>
`;
